mypy/types.py

```diff
@@ -1503,6 +1503,11 @@ def visit_type_type(self, t: TypeType) -> T:
     def visit_forwardref_type(self, t: ForwardRef) -> T:
         raise RuntimeError('Internal error: unresolved forward reference')
 
+    def visit_type_list(self, t: TypeList) -> T:
```

Comment:
Can other sorts of "synthetic" types show up in our traversal?

Would it be a better approach in the future to make the merger a synthetic type traverser?

Reply:
Hmm looks like we could be hitting other such types as well. Maybe making it a "synthetic type visitor" is the right thing to do. I'll consider making a follow-up PR about this. 

On the other hand, we should perhaps only have one sort of type visitor. I don't really see the benefit of having two sorts of visitors.